rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml

```diff
@@ -59,10 +85,10 @@
                 </hint>
             </perform>
             <where param="substitution">
-                <matches pattern="([a-zA-Z0-9]|-)*" />
+                <matches pattern="(?!rxjava2-starter$|http4-starter$|mongodb3-starter$|hdfs2-starter$|hdfs-starter$|mina2-starter$|mina-starter$|netty4-starter$|netty-starter$|netty4-http-starter$|netty-http-starter$|quartz2-starter$|quartz-starter$|jetty9-starter$|aws-starter$)([a-zA-Z0-9]|-)*" />
```

Comment:
I'm not sure about what is the intention of this regexp .....  
1. you are searching for this in the when &gt;artifact groupId="org.apache.camel" artifactId="camel-{substitution}-starter" &lt; but you are including word starter in the expression  
2. if I understand correctly the expression you say : (not these values at the end of the text) AND (any length of letter or number or dash ) so second group invalidates the first ?

Reply:
@jonathanvila I've just updated the regexp to be a little clearer. Let me know what you think, thanks for reviewing!